docs/source/userguide/plugin_development/algorithm.rst

```diff
@@ -0,0 +1,152 @@
+.. _add-an-algorithm:
+
+**********************************
+Extend GeoIPS with a new Algorithm
+**********************************
+
+To extend GeoIPS with a new ``algorithm`` plugin, first follow the :ref:`instructions for
+setting up a plugin package<plugin-development-setup>`.
+
+Module plugins are required to have several top-level variables:
+    * name
+    * interface
+    * family
+    * docstring
+
```

Comment:
@jsolbrig @AJThorpe @evrose54 Andrew said this documentation was much improved, and he could easily folow it to set up a new plugin.  Do you think we should get this merged as is, and we could open an additional issue to expand it to be more narrative style in the future?

Reply:
Good to go.